tools/gen_ort_dockerfile.py

```diff
@@ -387,60 +421,67 @@ def dockerfile_for_windows(output_file):
                         type=str,
                         required=False,
                         help='Home directory for TensorRT.')
+    
+    parser.add_argument('--common-repo-path',
+                        type=str,
+                        required=True,
+                        help='Path for add_dependencies.py.')
 
     FLAGS = parser.parse_args()
-
+    
     if target_platform() == 'windows':
         # OpenVINO EP not yet supported for windows build
         if FLAGS.ort_openvino is not None:
             print("warning: OpenVINO not supported for windows, ignoring")
             FLAGS.ort_openvino = None
 
-        # Default to CUDA based on CUDA_PATH envvar and TensorRT in
-        # C:/tensorrt
-        if 'CUDA_PATH'in os.environ:
-            if FLAGS.cuda_home is None:
-                FLAGS.cuda_home = os.environ['CUDA_PATH']
-            elif FLAGS.cuda_home != os.environ['CUDA_PATH']:
-                print("warning: --cuda-home does not match CUDA_PATH envvar")
+        if FLAGS.enable_gpu:
+            # Default to CUDA based on CUDA_PATH envvar and TensorRT in
+            # C:/tensorrt
+            if 'CUDA_PATH'in os.environ:
+                if FLAGS.cuda_home is None:
+                    FLAGS.cuda_home = os.environ['CUDA_PATH']
+                elif FLAGS.cuda_home != os.environ['CUDA_PATH']:
+                    print("warning: --cuda-home does not match CUDA_PATH envvar")
 
-        if FLAGS.cudnn_home is None:
-            FLAGS.cudnn_home = FLAGS.cuda_home
+            if FLAGS.cudnn_home is None:
+                FLAGS.cudnn_home = FLAGS.cuda_home
 
-        version = None
-        m = re.match(r'.*v([1-9]?[0-9]+\.[0-9]+)$', FLAGS.cuda_home)
-        if m:
-            version = m.group(1)
+            version = None
+            m = re.match(r'.*v([1-9]?[0-9]+\.[0-9]+)$', FLAGS.cuda_home)
+            if m:
+                version = m.group(1)
 
-        if FLAGS.cuda_version is None:
-            FLAGS.cuda_version = version
-        elif FLAGS.cuda_version != version:
-            print("warning: --cuda-version does not match CUDA_PATH envvar")
+            if FLAGS.cuda_version is None:
+                FLAGS.cuda_version = version
+            elif FLAGS.cuda_version != version:
+                print("warning: --cuda-version does not match CUDA_PATH envvar")
 
-        if (FLAGS.cuda_home is None) or (FLAGS.cuda_version is None):
-            print("error: windows build requires --cuda-version and --cuda-home")
+            if (FLAGS.cuda_home is None) or (FLAGS.cuda_version is None):
+                print("error: windows build requires --cuda-version and --cuda-home")
 
-        if FLAGS.tensorrt_home is None:
-            FLAGS.tensorrt_home = '/tensorrt'
+            if FLAGS.tensorrt_home is None:
+                FLAGS.tensorrt_home = '/tensorrt'
 
         dockerfile_for_windows(FLAGS.output)
 
     else:
```

Comment:
The nested if-else conditions are difficult to spot. How about making separate functions for setting up target windows and linux?
The `if FLAGS.enable_gpu` would be in those functions.